grakn-graql/src/main/java/ai/grakn/graql/internal/query/analytics/AbstractComputeQuery.java

```diff
@@ -171,7 +171,11 @@ void getAllSubTypes() {
 
     GraknComputer getGraphComputer() {
         if (graknComputer == null) {
-            graknComputer = Grakn.session(url, keySpace).getGraphComputer();
+            if(tx.isPresent()){
+                graknComputer = tx.get().session().getGraphComputer();
+            } else {
+                graknComputer = Grakn.session(url, keySpace).getGraphComputer();
+            }
```

Comment:
This class can probably be tidied some more. I don't think we need the `keySpace` and `url` fields for instance: they can just be retrieved from the `tx`.

I also suspect that this `else` should just throw. I think in all cases where the `tx` is absent, the `url` and `keyspace` will both be null (another reason to make them no longer fields).

Reply:
Ya I did not want to take that risk without checking with @duckofyork . Is it possible to rely only on the transaction here?

Reply:
Yes, we should remove `url` and `keySpace.`

Reply:
Then I will always use the provided transaction